### ISSUE TITLE
Support dependencies of issues through references

### ISSUE THREAD

**Issue by rarkins** [waffleio/waffle.io#803]
Seeing as you are already overriding one existing GitHub capability (i.e. labels) to generate the Kanban board, I have a major product feature request which would override another capability - issue references.

In my waffle board, I maintain a column after Backlog which is "Waiting". Issues in this column are there because they're waiting/blocked by another issue. Unfortunately, I still have to check and prune them regularly to see if they should move to the Ready column.

A killer waffle feature for me would be if you allow me to configure phrases such as "Waiting on #" or "Blocked by #" such that when those linked issues are closed, then waffle finds any "dependents" and moves them from Waiting to Ready if no other dependencies are open. This would help plug a big gap in using GitHub issues for project tracking too.

fixes #1996

**Comment by ashumz** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/alex@waffle.io/conversations/646922902)

"Simplified example:

I want to implement some feature in my application. This requires a new endpoint in my API and some UI additions, and I create two separate issues where the UI issue depends on the completion of the API issue.

As a project manager it is my task keep track of how the work progresses. This also means keeping an eye on when the API issues is completed so the UI issue can be moved to the Ready-column (manually).

The suggested Dependencies-feature with some visual indication in our waffle board along with the automove-to-ready-column-feature would let me focus on other things."

**Comment by zbeekman** [comment]
:+1:

Some sort of issue dependency syntax would be greatly appreciated. You could use ‘blocked’ or ‘requires’ or ‘depends’ to denote the issue dependency. Tasks like moving these issues from ‘backlog’ or ‘blocked’ column to a column like ‘ready' when all the dependencies have been resolved would be great. See also #595 #1125

**Comment by edgeprod** [comment]
+1

**Comment by BerkeleyTrue** [comment]
oh joy this PR
it would bring forth the rapture
to connect issues

**Comment by lorgio** [comment]
+1

**Comment by zanerock** [comment]
Product looks great, but basic dependency reference is hard requirement.

**Comment by tylerwgoza** [comment]
+1

**Comment by pnpritchard** [comment]
+1

**Comment by clbarnes** [comment]
Yes please! 'Blocked by', 'depends on' and maybe 'resolves' would be great!

**Comment by flimzy** [comment]
@clbarnes , "resolves" is already supported by standard GitHub issues, without any need for waffle.io integration.

**Comment by skotzko** [comment]
+1

**Comment by TomasMikula** [comment]
+1

**Comment by scottpickard** [comment]
+1

**Comment by katschre** [comment]
+1

**Comment by bstocks** [comment]
+1

**Comment by jhiver** [comment]
+1

**Comment by sjackman** [comment]
:+1:

**Comment by Lindenk** [comment]
+1

**Comment by jbenet** [comment]
Please, this would be huge for us!

**Comment by CandleInTheWind** [comment]
+1

**Comment by bmckie** [comment]
+1

**Comment by thel0r4x** [comment]
considering using codetree for huge issue dependency limitation in github:

https://codetree.com/

waffle doesn't do it.  jira and redmine can do both peer relationship and parent-child, of course but non-native.  

suggestions or comments on the above welcome.

**Comment by edgeprod** [comment]
Honestly, it's been so long, I'm going to mute and move on. This is an absolute MUST for development at Google, and there's been no mention of progress on it.

**Comment by syberphunk** [comment]
Perhaps someone like @burtonjc can comment.

**Comment by jbenet** [comment]
You can leverage @noffle's code here: https://github.com/noffle/github-dependency-crawl

more discussion relevant to people's interests here: https://github.com/jbenet/random-ideas/issues/37 (dependency graph viz and so on)

**Comment by leeogrady** [comment]
+1

**Comment by AlexSneedMiller** [comment]
Can anyone @waffleio comment if this is an issue that they want to work on, or if it's not on their roadmap for anytime in the near future?

**Comment by thel0r4x** [comment]
we ended up buying codetree access on top of github which works well.  other options exist.

anybody considering using github for something other than a smaller, tactical, dev-driven org would do well to consider limitations here.  doing it all over again we'd probably go with bitbucket.

**Comment by madmaze** [comment]
+1

**Comment by Jwata** [comment]
+1

**Comment by mauroservienti** [comment]
We'd like to have this one as well, even in the form of `Depends on #` similar to the `Connects to #` thing

**Comment by robertbray** [comment]
+1

**Comment by thel0r4x** [comment]
as a follow-up, below is a screen shot of how codetree handles dependencies in github:

![screen_shot_2017-05-04_at_15_10_09](https://cloud.githubusercontent.com/assets/20513825/25705029/1bb488ac-30dc-11e7-8bcb-b154b1bc607b.png)

it shows an issue 130 in a repo called "feature" (link filter profiles to labels).

this one issue that has both "needed by" and "needs" dependencies (2 and 1 github issues linked respectively below issue 130).

the "20" is a unit estimation and gets rolled up so you can track story points per milestone.  in my example, the children and parent issues (needed by and needs) don't have estimates but could.

allows a stackrank too and task board views (neither shown).

overall, it's been quite useful to us to continue using github with agile project tracking across releases.

**Comment by StellarDoor5319** [comment]
+1

I want this too 😀

**Comment by JonahMoses** [comment]
Out of curiosity, now that this issue has moved from `idea` to `planned` what does that mean for timelines? Curious because the issue was opened in 2014 and changed at the end of 2017.

**Comment by kellihsf** [comment]
Hi all, 👋 

Waffle now supports dependencies! Check out our [help docs here](https://help.waffle.io/dependencies) for more info. Email us at support@waffle.io with any comments or questions! 

![lt-dep-2_00000](https://user-images.githubusercontent.com/4775572/40389302-412d224e-5dcf-11e8-9f02-f210ed8a8ef7.gif)
